File: prjs/prog/prblm/projecteuler/Prob294.java
import java.math.BigInteger;
public class Prob294{
	public static void main(String args[])throws Exception{
		 BigInteger mul = BigInteger.ONE;
		 int pow = Integer.parseInt(args[0]);
		 BigInteger max = new BigInteger("10").pow(pow);
		 BigInteger nine = new BigInteger("9");
		 BigInteger t3 = new BigInteger("23");
		 System.out.println("guess "+max.divide(new BigInteger("414")));
		 long count=0;
                 for(BigInteger i=new BigInteger("10");mul.compareTo(max)<0;i=i.add(nine)){
			mul=i.multiply(t3);
			if(d(mul).compareTo(t3)==0){
                              //System.out.println("i-1 "+i.subtract(BigInteger.ONE).divide(nine));                
 				count = (count+1)%1000000000l;
			}
	         }
		 System.out.println(count);

             }

	static BigInteger d(BigInteger n){
		BigInteger sum = BigInteger.ZERO;
		BigInteger  ten = new BigInteger("10");
		while(n.compareTo(BigInteger.ZERO)>0){
			sum = sum.add(n.mod(ten));
			n = n.divide(ten);;
		}
		return sum;
	}
}

File: prjs/prog/prblm/projecteuler/Prob294.cpp
#include<iostream>
#include "bigint/BigInteger.hh"

long long d(long long n){
	long long sum=0;
	while(n>0){
		sum+= n%10;
                n/=10;
	}
	return sum;
}
int main(){
	long long eleven=11;
	long long tenpow[12]={1,10,100,1000,10000,100000,1000000,10000000,100000000,1000000000,10000000000,100000000000};
        long pown=1;
	BigInteger tenpowb = 10;
	for(int k=0;k<23;k++){
		cout<<"10^"<<pown<<"%23 : "<<(tenpowb%23)<<"\n";
		pown++;
		tenpowb*=10;
	}
	long long prod = 1;
	for(long long j=0;j<12;j++){
          prod*=eleven;
	}
	cout<<" N = "<<prod<<endl;
	long long n = 23;
	long long count = 0;
        long long mul = 0;
	int powcount=3;
        for(long long i=43478299;mul<10000000000;i++){
		mul = n*i;
		
	        long long sum = d(mul);
		if(mul<0){ cout<<mul<<endl; return 1;}

	        if(sum==23){
			if(mul>tenpow[powcount]){
				cout<<"S(n): "<<powcount<<" "<<count<<" "<<mul<<" "<<i<<endl;
				powcount++;
			}
		//cout<<i<<" "<<mul<<endl;
		count++;
		}
	}
	cout<<count<<endl;

}
